### PR TITLE
Add save_while_infer option to avoid storing too many results in memory

### LINKED ISSUE #61: Testing pipeline may fail to work when the testing dataset is too large due to memory issue
Current testing pipeline saves all results to RAM before writing to ROM. However, when the testing dataset is very large, such operation may fail

### PR DESCRIPTION
**Pull request by amjltc295**
## Why do we need this PR?
* To fix #61

## How is it implemented?
* Add save_while_infer option in template_test_config and update _update_output() in the tester to save results in each batch instead of putting everything in the epoch_output

### CODE REVIEW

File: src/worker/tester.py [hunk 1]
```diff
@@ -40,21 +46,39 @@ def _init_output(self):
...
+            # Clean previous results
+            epoch_output['saved'] = {k: [] for k in global_config.saved_keys}
+
+        for d in [data, model_output]:
+            update_epoch_output_from_dict(d)
+
+        if global_config.save_while_infer:
+            # Save results
+            name = self.data_loader.name
+            for i, output in enumerate(epoch_output['saved']['model_output']):
```
Comment (Nash2325138) [inline src/worker/tester.py]: There are multiple keys in `args.saved_keys`. Why only `model_output` is saved?

File: src/worker/tester.py [hunk 2]
```diff
@@ -40,21 +46,39 @@ def _init_output(self):
...
+                index = epoch_output['saved']['index'][i]
+                output_path = os.path.join(self.saving_dir, f'{name}_index{index:06d}.npy')
```
Comment (Nash2325138) [inline src/worker/tester.py]: If `len(args.saved_keys) > 1`, their filenames will be the same.

File: src/worker/tester.py [hunk 3]
```diff
@@ -40,21 +46,39 @@ def _init_output(self):
...
+                np.save(output_path, output)
+                if index % 1000 == 0:
+                    logger.info(f'Saving output {output_path}...')
```
Comment (Nash2325138) [inline src/worker/tester.py]: This output message will show up asynchronously with that of `_print_log`?

Comment (amjltc295) [reply src/worker/tester.py]: _print_log will only show the current batch num

File: src/worker/tester.py (hunk 2 above)
Comment (amjltc295) [reply src/worker/tester.py]: It is saved by the index; for now, only the model_output will be saved. This part should be customized by the user though

File: src/worker/tester.py (hunk 1 above)
Comment (amjltc295) [reply src/worker/tester.py]: Fixed

**Pull request merged**